sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferStore.scala

```diff
@@ -304,6 +304,38 @@ abstract class RapidsBufferStore(
       }
     }
 
+    override def getMemoryBuffer: MemoryBuffer = internalGetMemoryBuffer
+
+    override def getDeviceMemoryBuffer: DeviceMemoryBuffer = {
+      while (true) {
```

Comment:
Not sure if we need to limit the number of retries as done in the buffer catalog?

Reply:
Done.